### PR TITLE
Add config file and improve cli

### PR DESCRIPTION
**Pull request by cwaltken-edrans**
This pull requests introduces more verbose command-line flags as alternatives to the single letter options. Based on the more verbose flags you can now create an ini-style settings file storing your configuration for goad. By default `goad.ini` is being used if no other option is passed to the cli.

This pull request builds on the changes improving the cli suggested in the currently open pull request #22 by @matiaskorhonen. So that pull request could be closed on merge.

The new functionality is backwards compatible with the old cli and documented in the README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+  -h, --help                   Show context-sensitive help (also try --help-long
+                               and --help-man).
+  -u, --url=URL                URL to load test
+  -m, --method="GET"           HTTP method
+  -b, --body=BODY              HTTP request body
+  -c, --concurrency=10         Number of concurrent requests
+  -n, --requests=1000          Total number of requests to make
+  -N, --timelimit=3600         Seconds to max. to spend on benchmarking
+  -t, --timeout=15             Request timeout in seconds
+  -r, --region=REGION ...      AWS regions to run in (repeat flag to run in more
```
Comment (zeph) [inline README.md]: option `-r` is already used in AB, we shall move this into the INI files
(keep mentioned at the bottom of this output the default of 3 regions)

File: README.md [hunk 2]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+Flags:
+  -h, --help                   Show context-sensitive help (also try --help-long
+                               and --help-man).
+  -u, --url=URL                URL to load test
+  -m, --method="GET"           HTTP method
+  -b, --body=BODY              HTTP request body
+  -c, --concurrency=10         Number of concurrent requests
+  -n, --requests=1000          Total number of requests to make
+  -N, --timelimit=3600         Seconds to max. to spend on benchmarking
+  -t, --timeout=15             Request timeout in seconds
```
Comment (zeph) [inline README.md]: `-s` on AB

File: README.md [hunk 3]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+
+Flags:
+  -h, --help                   Show context-sensitive help (also try --help-long
+                               and --help-man).
+  -u, --url=URL                URL to load test
+  -m, --method="GET"           HTTP method
+  -b, --body=BODY              HTTP request body
+  -c, --concurrency=10         Number of concurrent requests
+  -n, --requests=1000          Total number of requests to make
+  -N, --timelimit=3600         Seconds to max. to spend on benchmarking
```
Comment (zeph) [inline README.md]: `-N` here would be `-t` on AB

File: README.md [hunk 4]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
-      HTTP Header to add to the request. This option can be set multiple times.
+$ goad --help
+usage: goad --url=URL [<flags>]
+
+An AWS Lambda powered load testing tool
+
+Flags:
+  -h, --help                   Show context-sensitive help (also try --help-long
+                               and --help-man).
+  -u, --url=URL                URL to load test
```
Comment (zeph) [inline README.md]: `-u` is a totally another feature in AB
please flatten this as to be the last parameter (without the flag) as in AB

File: README.md [hunk 5]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+  -m, --method="GET"           HTTP method
+  -b, --body=BODY              HTTP request body
```
Comment (zeph) [inline README.md]: `-b` POST body is a POST from file in AB

File: README.md [hunk 6]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+  -c, --concurrency=10         Number of concurrent requests
+  -n, --requests=1000          Total number of requests to make
+  -N, --timelimit=3600         Seconds to max. to spend on benchmarking
+  -t, --timeout=15             Request timeout in seconds
+  -r, --region=REGION ...      AWS regions to run in (repeat flag to run in more
+                               then one region)
+  -p, --awsprofile=AWSPROFILE  AWS named profile to use
```
Comment (zeph) [inline README.md]: please remove, as this is also not working properly, see issue https://github.com/goadapp/goad/issues/108

we can move it into the INI file if we prefer not to use the ENV variables

File: README.md [hunk 7]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+  -o, --output=OUTPUT          Optional path to JSON file for result storage
```
Comment (zeph) [inline README.md]: I'm not a fan of adding a new parameter.... but `-o` is used for something else in AB... `-g` and `-e` are used for a similar purpose, and `-j` is not defined... we might use it

File: README.md [hunk 8]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+  -H, --header=HEADER ...      HTTP request header (repeat flag to add more then
+                               one header)
+  -s, --settings=.goad         Load settings from file (defaults to .goad in
```
Comment (zeph) [inline README.md]: please don't... let's discuss parameters with me before taking these arbitrary decisions ... as `-s` is also an already used param in AB ... this has to be the default condition, no need to activate it explicitly, if there is a `default.ini` file in the working folder

File: README.md [hunk 9]
```diff
@@ -62,29 +62,57 @@ Goad will read your credentials from `~/.aws/credentials` or from the `AWS_ACCES
...
+
+```toml
+url = "http://example.com/"
+timeout = 15
+concurrency = 10
+requests = 100
+timelimit = 60
+method = "GET"
+body = "Hello world"
+regions = ["eu-west-1", "us-east-1"]
```
Comment (zeph) [inline README.md]: not the format we discussed, please check https://github.com/zeph/goad/blob/master/desenmerda-te/default.ini

**Comment by zeph** [comment]
please check issue https://github.com/goadapp/goad/issues/117 too

File: README.md (hunk 1 above)
Comment (cwaltken-edrans) [inline README.md]: Okay, good point. I'll remove the short -r fag and keep only the long version to avoid conflict with AB.

File: README.md (hunk 4 above)
Comment (cwaltken-edrans) [inline README.md]: I'll remove the -u flag completely and replace it with an argument passed to goad just as in AB.

File: README.md (hunk 5 above)
Comment (cwaltken-edrans) [inline README.md]: I'll remove the short flag -b to be compatible with AB. To have a POST from file would require implementation.

File: README.md (hunk 8 above)
Comment (cwaltken-edrans) [inline README.md]: Okay, i'll remove the parameter.

File: README.md (hunk 9 above)
Comment (cwaltken-edrans) [inline README.md]: I'll first change the cli. Then I'll adopt the reference format.

**Pull request closed without merging**